Sources/edge-helper/Services/IPAddressManager+macos.swift

```diff
@@ -0,0 +1,216 @@
+#if os(macOS)
```

Comment:
This should be generic enough to work on linux too. the caveat here is the Darwin part. we should abstract away and plug that in via some protocol instead.